spec/unit/installer/user_project_integrator/target_integrator_spec.rb

```diff
@@ -366,6 +367,105 @@ module Pod
             ${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/ReleaseVendoredFramework.framework
           )
         end
+
+        it 'adds a custom shell script phase' do
+          @pod_bundle.target_definition.stubs(:script_phases).returns([:name => 'Custom Script', :script => 'echo "Hello World"'])
+          @target_integrator.integrate!
+          target = @target_integrator.send(:native_targets).first
+          phase = target.shell_script_build_phases.find { |bp| bp.name == @user_script_phase_name }
+          phase.name.should == '[CP] Custom Script'
+          phase.shell_script.should == 'echo "Hello World"'
+        end
+
+        it 'removes outdated custom shell script phases' do
+          @pod_bundle.target_definition.stubs(:script_phases).returns([:name => 'Custom Script', :script => 'echo "Hello World"'])
+          @target_integrator.integrate!
+          target = @target_integrator.send(:native_targets).first
+          target.shell_script_build_phases.find { |bp| bp.name == @user_script_phase_name }.should.not.be.nil
+          @pod_bundle.target_definition.stubs(:script_phases).returns([])
+          @target_integrator.integrate!
+          target.shell_script_build_phases.find { |bp| bp.name == @user_script_phase_name }.should.be.nil
+        end
+
+        it 'moves custom shell scripts to their correct index' do
+          shell_script_one = { :name => 'Custom Script', :script => 'echo "Hello World"' }
+          shell_script_two = { :name => 'Custom Script 2', :script => 'echo "Hello Aliens"' }
+          @pod_bundle.target_definition.stubs(:script_phases).returns([shell_script_one, shell_script_two])
+          @target_integrator.integrate!
+          target = @target_integrator.send(:native_targets).first
+          target.shell_script_build_phases.map(&:name).should == [
+            '[CP] Check Pods Manifest.lock',
+            '[CP] Embed Pods Frameworks',
+            '[CP] Copy Pods Resources',
+            '[CP] Custom Script',
+            '[CP] Custom Script 2',
+          ]
+          shell_script_one_uuid = target.shell_script_build_phases[3].uuid
+          shell_script_two_uuid = target.shell_script_build_phases[4].uuid
+          @pod_bundle.target_definition.stubs(:script_phases).returns([shell_script_two, shell_script_one])
+          @target_integrator.integrate!
+          target.shell_script_build_phases.map(&:name).should == [
+            '[CP] Check Pods Manifest.lock',
+            '[CP] Embed Pods Frameworks',
+            '[CP] Copy Pods Resources',
+            '[CP] Custom Script 2',
```

Comment:
Can we also prefix these as being "User" scripts to avoid breaking changes and conflicts when we add pod spec support for this?

Reply:
Hm good idea what would a good prefix be?

Reply:
Maybe `[CP-User]`?

Reply:
No real preference tbh

Reply:
👍 will update

Reply:
@DanToml updated.